docs/source/development/controllers.rst

```diff
@@ -0,0 +1,221 @@
+:meth:`evolver.controller.Controller.control`, code can read sensor values (e.g.
+by calling :meth:`evolver.hardware.interface.SensorDriver.get`) and effect the
+environment based on read sensor values (e.g. by calling
+:meth:`evolver.hardware.interface.EffectorDriver.set` with appropriate input).
+
+As with other eVolver components, controllers contain `Config` classes that
+define the configuration options for the experiment, which can be read from
+on-disk config and modified via the API and UIs. authors should provide a
+`Config` class that inherits from :class:`evolver.controller.Controller.Config`
+with appropriate options for the experiment. See :doc:`config` for
+more details on configuration.
+
+Getting started
+---------------
+
+A breif example will help illustrate what is required to implement a new
+controller. In the example below, we define a simple experiment::
+
+    from evolver.controller import Controller
+    from evolver.hardware.standard import Temperature
+
+    class MyController(Controller):
+        class Config(Controller.Config):
+            limit: int = 42
+            hysteresis: int = 2
+
+        def control(self):
+            for vial in self.vials:
+                density = self.evolver.hardware['temp'].get[vial].density
+                if density > self.limit:
+                    self.evolver.hardware['temp'].set(Temperature.Input(vial=vial, value=30))
+                elif density < self.limit - self.hysteresis:
+                    self.evolver.hardware['temp'].set(Temperature.Input(vial=vial, value=38))
+
+In the above you can note that all the business logic is contained within the
+`control` method, while configuration parameters are available as attributes of
+the controller (e.g. `self.limit` and `self.hysteresis`). Further, the
+controller gains access to hardware state via the `evolver` property, which is
+an instance of :py:class:`evolver.device.Evolver`, and can be used to access
+hardware drivers (e.g. `self.evolver.hardware['temp']`) to get current state and
+set desired state.
+
+.. note::
+
+    The `evolver` property comes from hookup to the device (which happens for
+    example in creating a new eVolver from configuration), see also
+    :ref:`evolver-hookup` below. `vials` comes from the base controller `Config`
+    class, and represents a list of vials to operate on.
+
+Support multiple vial configurations
+------------------------------------
+
+You might note in the above example that:
+
+* We loop over a set of vials in the controller, and
+* We have scalar values for the config parameters, not an array of values for
+  each vial.
+
+The concept here is that each instance of a controller is a specific instance of
+the experiment on a set of vials. In the simplest case one experiment with a
+given set of configuration parameters is run on all vials on the box. The
+eVolver accepts a list of controllers, however, which can represent either
```

Comment:
I think it should read either:
"controllers, which, ..."
or
"controllers, however, these..."
I think...